20 create student page

SukhanovPavel opened the pull request:
fix: 
add css modules
add pattern for tg nickname input
add store for user and user settings
delete 'onChange' in input
delete style prop, delete comments
add ESLint

SukhanovPavel committed:
add student page and student settings page

SukhanovPavel committed:
feat(compponents) create student page and student settings page

SukhanovPavel committed:
test

SukhanovPavel committed:
test

SukhanovPavel committed:
fix(components): add css modules, add pattern for tg add store for user and user settings

add css module
add pattern for tg nickname input
add store for user and user settings
delete
'onChange' in input
delete style prop
delete comments
add ESLint

META DATA: Create student page

Hellomyguest commented on src/pages/student/page.tsx:
```diff
@@ -1,11 +1,74 @@
+            <Link to={`https://t.me/${userSettings.tgNickName.slice(1)}`} ><u>tg: {userSettings.tgNickName || 'nickname'}</u></Link>
+            <Link to={`https://github.com/${userSettings.githubNickName}`} ><u>github: {userSettings.githubNickName || 'nickname'}</u></Link>
+            <Link to={routes.studentSettings} ><u>{userSettings.htmlRepository || 'repository html'}</u></Link>
+            <Link to={routes.studentSettings} ><u>{userSettings.reactRepository || 'repository react'}</u></Link>
```
Link to={`https://github.com/${userSettings.githubNickName}/${userSettings.htmlRepository}`}
Link to={`https://github.com/${userSettings.githubNickName}/${userSettings.reactRepository}`}

Hellomyguest commented on src/pages/student/page.tsx:
```diff
@@ -1,11 +1,74 @@
+    <Content className={styles._}>
+      <Row>
+        <Space size={100}>
+          <Col span={50}>
```
span max 24, maybe it's better to use one col for table and the second for user links?

Hellomyguest commented on src/pages/student/page.tsx:
```diff
@@ -1,11 +1,74 @@
+                </Card.Grid>
+              ))}
+            </Card>
+            <Row gutter={20}>
```
Table beside Card.Grid?

Hellomyguest commented on src/pages/student/model.ts:
```diff
@@ -0,0 +1,47 @@
+
+export const setUserSettings = createEvent<UserSettings>();
+
+$userSettings.on(setUserSettings, ( _, payload ) => payload)
```
maybe this way is better:
![image](https://user-images.githubusercontent.com/48411208/210654253-a50e6fd9-9c5a-40dd-bb82-f720c7fed0df.png)

Hellomyguest commented on src/pages/student/settings/page.tsx:
```diff
@@ -1,11 +1,137 @@
+      <Form
+        name="userSettings"
+        className="Form"
+        initialValues={{ remember: true }}
```
initialValues={userSettings} and then delete all the input values props

SukhanovPavel committed:
fix(components): fix user store

fix user store
fix ESLint
add initial values on userSettings page

META DATA: 20 create student page

SukhanovPavel committed:
Merge branch 'develop' into 20-create-student-page

Hellomyguest commented on src/pages/student/page.module.css:
```diff
@@ -0,0 +1,40 @@
+}
+
+.cardBody {
+  width: 50rem;
```
why?
screen size 1024px -  settings broke

Hellomyguest commented on src/pages/student/page.tsx:
```diff
@@ -1,11 +1,83 @@
+              <Row gutter={20}>
+                {homeworkCards.map((card) => (
+                  <Col key={card.deadline} className={styles.gridStyle}>
+                    <div>{card.homeworkStatus ? "" : card.deadline}</div>
```
why div? won't it be better to use one homeworksCards.map => card with content in grid system?
[https://codesandbox.io/s/7r6080?file=/demo.tsx:249-261](url)

Hellomyguest commented on src/pages/student/settings/page.tsx:
```diff
@@ -1,11 +1,105 @@
+            rules={[
+              {
+                required: true,
+                message: "Please input name of repository!",
```
if i just register and don't have repositories yet - i can't fill this.

Hellomyguest commented on src/pages/student/settings/page.tsx:
```diff
@@ -1,11 +1,105 @@
+            rules={[
+              {
+                required: true,
+                message: "Please input name of repository!",
```
if i just register and don't have repositories yet - i can't fill this.

Hellomyguest commented on src/pages/student/page.tsx:
```diff
@@ -1,11 +1,83 @@
+              <Title level={3}>Оценки за дз</Title>
+              <Card className={styles.cardBody}>
+                {homeworkCards.map((card) => (
+                  <Card.Grid key={card.deadline} className={styles.gridStyle}>
```
to understand that homework is acceptet u can do it like this: 
<Card.Grid key={card.deadline} className={`${styles.gridStyle} ${card.homeworkStatus ? styles.green : null}`}>

but in our case we also have yellow one, so make enum.
also if we need 2 or more classes i think it's better to use classnames plugin

Hellomyguest commented on src/pages/student/page.tsx:
```diff
@@ -1,11 +1,83 @@
+                    // type={card.homeworkStatus}
+                    >
+                      {card.homework}
+                    </Text>
```
to understand that homework is accepted u can do it like this:
![image](https://user-images.githubusercontent.com/48411208/211222960-97b6973e-2658-4320-9405-c4d48b7d51cf.png)

but in our case we also have yellow one, so make enum.

dpolevodin commented on src/pages/student/model.ts:
```diff
@@ -0,0 +1,48 @@
+
+$user.on(setUserSettings, (state, payload) => {
+  const user = { ...state };
+  user.settings = payload;
```
можно попробовать заменить более короткой записью
```
$user.on(setUserSettings, (state, payload) => ({ ...state, settings: payload }))
```

dpolevodin commented on src/pages/student/settings/mocks.ts:
```diff
@@ -0,0 +1,32 @@
+export type HomeworkCards = {
```
Если тип для одной карточки лучше использовать название```HomeworkCard```

dpolevodin commented on src/pages/student/settings/page.tsx:
```diff
@@ -1,11 +1,105 @@
+    toStudentPage,
+  ]);
+  const handleFinish = (value: UserSettings) => {
+    setUserSettingsFn(value);
```
Предлагаю попробовать завязаться на ```setUserSettingsFn``` в логике эффектора для редиректа
Мне кажется, можно избежать дополнительного вызова функции ```toStudentPageFn``` в данном хэндлере

SukhanovPavel committed:
refactor(components): divided into small components

divided into small components
connected the store from the admin model

SukhanovPavel committed:
refactor(components):

SukhanovPavel committed:
Merge branch 'develop' into 20-create-student-page
